howdoi/howdoi.py

```diff
@@ -97,8 +97,27 @@ def _get_result(url):
         raise e
 
 
+def _add_links_to_text(element):
+    duplicate_element = element.clone()
+    hyperlinks = duplicate_element.find('a')
+
+    for hyperlink in hyperlinks:
+        href = hyperlink.attrib['href']
+        copy = hyperlink.text
+        if (copy == href):
+            replacement = copy
+        else:
+            replacement = "[{0}]({1})".format(copy, href)
+        _replace_first_link(duplicate_element, replacement)
```

Comment:
This is clever. I was trying to think of a case where it would fail and can't seem to come up with anything. However, is it possible to simply to call `.replace_with` on the `hyperlink` object within the `for` loop rather than assuming that it will be the first `a` found on the parent element?

Reply:
You can only call ```.replace_with``` as a pquery object. In order to get the copy and the ```href``` of the ```hyperlink```, I have to unwrap the pquery object, thereby losing access to the ```.replace_with``` method. When I refactored the code, I "wrapped" the hyperlink back up into a pquery object, so I could then call ```.replace_with``` directly on the element. Still hacky, but less hacky than the old method.